### DISCUSSION TITLE
How to show teasers of blog posts on blog category page, aggregated by tag?

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#787] [Ask a question] [answered]
Dears,

after a longer absence, I returned to my most beloved CMS - Yellow, starting anew with a bigger project, and I have run into a crucial problem.

I plan to have several category pages on the site, consisting of teasers (content from blog-start pages) and links to the blog pages.

I want these category pages to show only certain blog post teasers, so is it possible to **show only blog posts relating to topic A, topic B etc- depending on a tag found in the blog posts?**

I have seen the "BlogWithFilter" page settings, but have no clue what this does.

Apart from that, I am aware of the phenomenal **Include** extension, which helps me with the same task on Default (= non-blog) pages, but I do not want to manually link the blog posts on blog-start pages - it would be nice if this could be done automatically via tags.

Any help greatly appreciated!

Cheers,

JetGum

**Comment by schulle4u** [comment, marked as answer]
Welcome back! 🙂
Thanks to the API you can easily filter blog pages by tag in your `system/layouts/blog-start.html`. Here is an example layout which only displays blog pages filed under the news tag, have a closer look at the first `foreach` line. 

```
<?php $this->yellow->layout("header") ?>
<div class="content">
<div class="main" role="main">
<?php if ($this->yellow->page->get("blogWithFilter")): ?>
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<?php endif ?>
<?php foreach ($this->yellow->page->getPages("blog")->filter("tag", "news")->paginate($this->yellow->system->get("blogPaginationLimit")) as $page): ?>
<?php $page->set("entryClass", "entry") ?>
<?php if ($page->isExisting("tag")): ?>
<?php foreach (preg_split("/\s*,\s*/", $page->get("tag")) as $tag) { $page->set("entryClass", $page->get("entryClass")." tag-".$this->yellow->toolbox->normaliseArguments($tag, false)); } ?>
<?php endif ?>
<div class="<?php echo $page->getHtml("entryClass") ?>">
<div class="entry-title"><h1><a href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("title") ?></a></h1></div>
<div class="entry-meta"><p><?php echo $page->getDateHtml("published") ?> <?php echo $this->yellow->language->getTextHtml("blogBy") ?> <?php $authorCounter = 0; foreach (preg_split("/\s*,\s*/", $page->get("author")) as $author) { if (++$authorCounter>1) echo ", "; echo "<a href=\"".$this->yellow->page->getLocation(true).$this->yellow->toolbox->normaliseArguments("author:$author")."\">".htmlspecialchars($author)."</a>"; } ?></p></div>
<div class="entry-content"><?php echo $this->yellow->toolbox->createTextDescription($page->getContent(), 0, false, "<!--more-->", "<a href=\"".$page->getLocation(true)."\">".$this->yellow->language->getTextHtml("blogMore")."</a>") ?></div>
</div>
<?php endforeach ?>
<?php $this->yellow->layout("pagination", $this->yellow->page->getPages("blog")) ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

I'm not sure if and how its possible to filter by more than one tag at the same time, but this should give you a starting point. Happy coding!

Steffen

**Reply by JetGum** [reply]
Steffen

wow!!! This was fast and it works like a charm... (for one category page, though...). 

How can I make this work for multiple category pages with different tags? Do I need to create a different blog-start.html for each category page or is it possible to make this more "dynamic"? Would be great to have a tag in the category page and then have the "->filter("tag", "tag1")" part of blog-start,html to look dynamically for the tag of the category page and then take this tag and aggregate all pertaining blog posts? Hope it's clear what i mean :-)

That would be awesome!

Cheers and thx a million!

Carsten

**Reply by JetGum** [reply]
Dears,

I copied and renamed the blog-start.html file, because I thought I could simply add another layout file with a different tag filter.
But when I edit the Page Settings Layout to, for example, blog-start-1.html, it does not work. 
Error Message: YellowCore::sendPage layout:error

What am I missing?

Cheers,

Carsten

**Reply by schulle4u** [reply]
Afaik you can only have one blog-start layout file. Not sure if this would help you, but you could try to implement a set of redirect pages for your tags. As an example content structure: 

```
* content/
  - 1-home/
  - 2-redirect-to-tag1/
  - 3-redirect-to-tag2/
  - blog/
```

The blog folder contains regular blog pages and uses the blog-start layout, but is not visible in the navigation. Instead you have simple redirect pages which bump your visitors in the right direction. An example redirect page for `content/2-redirect-to-tag1/page.md` would then look like: 

```
---
Title: Redirecting to tag1
Redirect: /blog/tag:tag1/
---
```

Bear in mind that the redirects only work outside editing mode. Perhaps a little complicated, but this would save you from layout modifications. 

Steffen

**Reply by JetGum** [reply]
Hi Steffen,

thx for this interesting input. I will give the redirect idea a try over the weekend. 

To get a general idea of what I am trying to accomplish: The structure of the website is that all content files are in root (= 1-home), and other content folders merely have a page.md in blog-start format to aggregate the content along certain tags (coffee, tea, cake, fruit etc.).

Another question_ So it isn't possible to rename any layout file and re-use it to have a different layout on a page/blog page?
Another idea: Can you have multiple blogs on one website?

Cheers,

Carsten

**Reply by JetGum** [reply]
Hi Steffen,

thanks for your support, I really could not get the redirects to work properly, so I came up with another solution within the blog-framework.

1. My homepage uses blog-start.html
Here, I use your excellent tag-filter in the layout file. 
All blog posts that I want to appear as teasers on the homepage have the tag:homepage.

2. All other category pages (shown in the navigation) are blog pages with individual layout files (topic-a,html, topic-b.html etc.).

Cheers,

Carsten
